src/ucm/util/sys.c

```diff
@@ -299,7 +299,7 @@ void ucm_prevent_dl_unload()
      * NODELETE flag to the dynamic link map.
      */
     (void)dlerror();
-    dl = dlopen(info.dli_fname, RTLD_LOCAL|RTLD_LAZY|RTLD_NODELETE);
+    dl = dlopen(info.dli_fname, RTLD_LOCAL|RTLD_NOW|RTLD_NODELETE);
```

Comment:
IMO, RTLD_NOW can incur some overhead of resolving all symbols
Can we avoid it?

Reply:
here is opening already loaded module (libucm.so) - RTLD_NOW will do nothing

Reply:
```
       If the same library is loaded again with dlopen(), the same file handle is returned.  The dl library  maintains  reference  counts
       for  library  handles, so a dynamic library is not deallocated until dlclose() has been called on it as many times as dlopen() has
       succeeded on it.  The _init() routine, if present, is called only once.  But a subsequent call with RTLD_NOW may force symbol res‐
       olution for a library earlier loaded with RTLD_LAZY.
```

Reply:
are you talking about ```But a subsequent call with RTLD_NOW may force symbol resolution for a library earlier loaded with RTLD_LAZY.```???

yes, it may affect, but this function is called when some (or even most) of symbols are resolved by direct calls and impact from dlopen(RTLD_NOW) should be minimal

Reply:
one of the flags RTLD_NOW|RTLD_LAZY must be set to call dlopen & I don't see ways to make different way :(

Reply:
what is the reason LAZY did not work?

Reply:
in short:
```
#0  0x00002ba275633bd6 in _dl_relocate_object () from /lib64/ld-linux-x86-64.so.2
.......
#16 <signal handler called>
#17 0x00002ba275633bd6 in _dl_relocate_object () from /lib64/ld-linux-x86-64.so.2
.......
#24 0x0000000077168041 in dlopen@@GLIBC_2.2.5 () from /lib64/libdl.so.2
#25 0x00002ba28d6d9714 in ucm_prevent_dl_unload () at util/sys.c:304
.......
```
it seems issue in loader: in case if LAZY flag is set loader assumes that pointers are not initialized... RTLD_NOW resolves issue

Reply:
in which situation RTLD_LAZY segfault? can we identify this case and use RTLD_NOW only then?